tests/cpp/test_move_pad.cpp

```diff
@@ -0,0 +1,437 @@
+
+  FusionExecutorCache fec(std::move(fusion));
+  auto out_tensors = fec.runFusionWithInputs(aten_inputs);
+
+  FusionKernelRuntime* runtime = fec.getMostRecentKernelRuntime();
+  EXPECT_EQ(runtime->fusionSegments()->groups().size(), 1);
+
+  testValidate(fec.fusion(), out_tensors, aten_inputs, __LINE__, __FILE__);
+}
+
+TEST_F(MovePadTest, BinaryCat) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* tv0 = makeContigConcreteTensor({4, 10});
+  TensorView* tv1 = makeContigConcreteTensor({4, 10});
+  TensorView* tv2 = makeContigConcreteTensor({2, 10});
+  TensorView* tv3 = add(tv0, tv1);
+  TensorView* tv4 = cat({tv2, tv3}, /*dim=*/0);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addInput(tv2);
+  fusion->addOutput(tv4);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({4, 10}, options);
+  at::Tensor t1 = at::randn({4, 10}, options);
+  at::Tensor t2 = at::randn({2, 10}, options);
+  std::vector<c10::IValue> aten_inputs = {t0, t1, t2};
+
+  FusionExecutorCache fec(std::move(fusion));
+  auto out_tensors = fec.runFusionWithInputs(aten_inputs);
+
+  FusionKernelRuntime* runtime = fec.getMostRecentKernelRuntime();
+  EXPECT_EQ(runtime->fusionSegments()->groups().size(), 1);
+
+  testValidate(fec.fusion(), out_tensors, aten_inputs, __LINE__, __FILE__);
+}
+
+TEST_F(MovePadTest, BinaryBroadcastOnNonCatDim) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* tv0 = makeContigConcreteTensor({4, 10});
+  TensorView* tv1 = makeContigConcreteTensor({10});
+  // broadcast on non-pad dimension should propagate across binary operations
+  TensorView* tv1_b = broadcast(tv1, {true, false});
+  TensorView* tv2 = makeContigConcreteTensor({4, 5});
+  // tv1_b [bS, iS]. pad axes [1,] is not on broadcast dimension, we should be
+  // able to propagate pad
+  TensorView* tv3 = add(tv0, tv1_b);
+  TensorView* tv4 = cat({tv2, tv3}, /*dim=*/1);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addInput(tv2);
+  fusion->addOutput(tv4);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({4, 10}, options);
+  at::Tensor t1 = at::randn({10}, options);
+  at::Tensor t2 = at::randn({4, 5}, options);
+  std::vector<c10::IValue> aten_inputs = {t0, t1, t2};
+
+  FusionExecutorCache fec(std::move(fusion));
+  auto out_tensors = fec.runFusionWithInputs(aten_inputs);
+
+  FusionKernelRuntime* runtime = fec.getMostRecentKernelRuntime();
+  EXPECT_EQ(runtime->fusionSegments()->groups().size(), 2);
+
+  // ensure that we propagate the pad across binary operation and the first
+  // segment is no-op
+  auto scheduler = runtime->schedulerHeuristics()->heuristicsList().at(0).get();
+  auto heuristic = scheduler->heuristic();
+  NVF_CHECK(
+      heuristic == ScheduleHeuristic::NoOp,
+      "Unexpected heuristic: ",
+      heuristic);
```

Comment:
We can probably simplify this using UnorderedElementsAre and HeuristicIs, e.g., https://github.com/NVIDIA/Fuser/blob/4314090a6ab68b955220fcced675b5e05aa9cf0e/tests/cpp/test_alias.cpp#L1111-L1115